backend/pick-git/src/main/java/com/woowacourse/pickgit/user/application/UserService.java

```diff
@@ -117,4 +124,23 @@ private void validateDifferentSourceTarget(User source, User target) {
             throw new SameSourceTargetUserException();
         }
     }
+
+    public List<UserSearchResponseDto> searchUser(AppUser appUser, UserSearchRequestDto request) {
+        Pageable pageable = PageRequest.of(request.getPage(), request.getLimit());
+        List<User> users = userRepository.findAllByUsername(request.getKeyword(), pageable);
+
+        if (appUser.isGuest()) {
+            return users.stream()
+                .map(user -> new UserSearchResponseDto(user.getImage(), user.getName(), null))
+                .collect(toList());
+        }
+
+        User loginUser = findUserByName(appUser.getUsername());
+
+        return users
+            .stream()
+            .filter(user -> !user.equals(loginUser))
+            .map(user -> new UserSearchResponseDto(user.getImage(), user.getName(), loginUser.isFollowing(user)))
+            .collect(toList());
```

Comment:
```suggestion
        return users
            .stream()
            .filter(user -> !user.equals(loginUser))
            .map(user -> new UserSearchResponseDto(
                    user.getImage(),
                    user.getName(),
                    loginUser.isFollowing(user)
                )
            )
            .collect(toList());
```

컨벤션이 맞는건진 모르겠지만 100줄 넘어가서 정렬했습니다.